### ISSUE TITLE
findn() causes generated function body is not pure error

### ISSUE THREAD

**Issue by AzamatB** [JuliaLang/julia#24321]
Consider the code:
``` Julia
julia> findn(trues(2,2,2))
ERROR: generated function body is not pure. this likely means it contains a closure or comprehension.
```

**Comment by ararslan** [comment]
Looks like this works on 0.5 but not on 0.6 or master.

**Comment by AzamatB** [comment]
`findn()` was superseded by `findall()`, which now seems to give the expected output:
```Julia
julia> findall(trues(2,2,2))
8-element Array{CartesianIndex{3},1}:
 CartesianIndex(1, 1, 1)
 CartesianIndex(2, 1, 1)
 CartesianIndex(1, 2, 1)
 CartesianIndex(2, 2, 1)
 CartesianIndex(1, 1, 2)
 CartesianIndex(2, 1, 2)
 CartesianIndex(1, 2, 2)
 CartesianIndex(2, 2, 2)
```
So I am not if this issue is still relevant.

**Comment by AzamatB** [comment]
This was fixed in 0.7.
